devito/arch/compiler.py

```diff
@@ -410,9 +412,15 @@ def __init__(self, *args, **kwargs):
                 self.ldflags += ['-march=%s' % platform.march]
         elif platform is M1:
             # NOTE:
-            # -march=native unsupported
-            # openmp unusable
-            pass
+            # This requires to install Aplle's llvm (through brew) to install the openmp
+            # librairies and c++ includes. Once installed, both Apple's default clang and
+            # howebrew clang support openmp. Check that openmnp lib exist.
+            if path.exists('/opt/homebrew/opt/llvm/lib/libomp.dylib'):
+                self.ldflags += ['-mcpu=apple-m1', '-fopenmp']
+                self.ldflags += ['-L/opt/homebrew/opt/llvm/lib']
+            else:
+                warning("Openmp librairies not foud. Install homebrew llvm:"
```

Comment:
this warning could also be moved inside the aforementioned `get_clang_path` (or `get_llvm_path` FWIW)

Optionally, you could also return a dict from this function -- one entry for the include dir, one entry for the libs (where you would find `libomp.dylib`), etc